scheduler.go

```diff
@@ -32,22 +38,26 @@ func (s *Scheduler) StartBlocking() {
 
 // StartAsync starts a goroutine that runs all the pending using a second-long ticker
 func (s *Scheduler) StartAsync() chan struct{} {
-	stopped := make(chan struct{})
-	ticker := s.time.NewTicker(1 * time.Second)
+	if s.running {
+		return s.stopChan
+	}
+	s.running = true
 
+	ticker := s.time.NewTicker(1 * time.Second)
 	go func() {
 		for {
 			select {
 			case <-ticker.C:
 				s.RunPending()
-			case <-stopped:
+			case <-s.stopChan:
 				ticker.Stop()
+				s.running = false
 				return
 			}
 		}
 	}()
 
-	return stopped
+	return s.stopChan
```

Comment:
(Unrated to this PR) Do we really need to expose this `chan` to the user? We already have a `s.Stop()` that takes care of stopping the scheduler.

Reply:
You're absolutely right, but as this would a breaking change I decided to leave it for another PR